outputs.tf

```diff
@@ -11,7 +11,7 @@ output "private_dns" {
 }
 
 output "public_dns" {
-  value = "${aws_instance.default.public_dns}"
+  value = "${coalesce(join("", null_resource.eip.*.triggers.public_dns), aws_instance.default.public_dns)}"
```

Comment:
`coalesce(concat(....` seems like the more appropriate pattern.

Reply:
I've tested different variants of output and have luck only with `join()`

```terraform
output test {
  value = "${coalesce(join("", null_resource.eip.*.triggers.test), var.eip)}"
}

output test1 {
  value = "${coalesce(concat(null_resource.eip.*.triggers.test), var.eip)}"
}

output test3 {
  value = "${coalesce(concat("", null_resource.eip.*.triggers.test), var.eip)}"
}
```
```sh
Apply complete! Resources: 1 added, 0 changed, 1 destroyed.

Outputs:

test = ec2-91-23-33-14.us-central-1.compute.amazonaws.com
```